### PR TITLE
Update UG/DG for update command

### PR DESCRIPTION
**Pull request by dvdweien**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -206,6 +206,64 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+The update mechanism is facilitated by `PocketEstate`. It implements the update feature with 3 parts, the `Status` field in `Property`, a `UpdateCommandParser` and `UpdateCommand`.
+
+The `Status` field consists of a `Status` interface with a `next()` method that returns a `Status` representing the next stage of the property selling process. There are 3 classes that implement `Status`, `Option`, `SalesAgreement` and `Completion`. Each class takes in an `Offer` which represents the price that was offered for the property.
+
+* `Option` — Represents the stage where the buyer exercises the Option to Purchase.
+* `SalesAgreement` — Represents the stage where the buyer is considering the Sales and Purchase Agreement.
+* `Completion` — Represents the stage where the property has been sold.
+
+(insert class diagram of status)
```
Comment (candyhy) [inline docs/DeveloperGuide.md:221]: are your diagrams to be inserted later?

**Review by candyhy — COMMENTED** [review]
LGTM with some comments

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -206,6 +206,64 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+The following activity diagram summarizes what happens when a user executes an `UpdateCommand`:
+
+(insert overall update activity diagram here)
+
+(insert UpdateCommandParser activity diagram here)
+
+The following sequence diagram shows how the update operation works:
+
+(insert update new activity diagram here)
```
Comment (candyhy) [inline docs/DeveloperGuide.md:265]: these too

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -206,6 +206,64 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+(some object diagram?)
+
+Step 5. The user then decides that having the `Completion` status on the first property was a mistake and executes the command `update 1 u/cancel`.
+
+(some object diagram?)
+
+
+#### Design consideration:
+
+##### Aspect: How undo executes
```
Comment (candyhy) [inline docs/DeveloperGuide.md]: should this be 'undo'?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -206,6 +206,64 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+(some object diagram?)
+
+Step 4. The user executes `update 1 u/proceed` to move the `Status` of the first property to `Completion`.
+
+(some object diagram?)
+
+Step 5. The user then decides that having the `Completion` status on the first property was a mistake and executes the command `update 1 u/cancel`.
+
+(some object diagram?)
```
Comment (candyhy) [inline docs/DeveloperGuide.md:249]: are these to be added later?

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (dvdweien) [reply docs/DeveloperGuide.md:221]: yeah, yet to learn plantUML will add next week

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (dvdweien) [reply docs/DeveloperGuide.md:265]: yeah, these too

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (dvdweien) [reply docs/DeveloperGuide.md]: Nice catch! thanks will change it

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (dvdweien) [reply docs/DeveloperGuide.md:249]: yepp!

**Review by candyhy — APPROVED** [review]
LGTM

**Pull request merged**